Reply/app/src/main/java/com/example/reply/ui/ReplyListContent.kt

```diff
@@ -0,0 +1,185 @@
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.reply.ui
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyListState
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Edit
+import androidx.compose.material3.Icon
+import androidx.compose.material3.LargeFloatingActionButton
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.example.reply.R
+import com.example.reply.data.Email
+import com.example.reply.ui.components.EmailDetailAppBar
+import com.example.reply.ui.components.ReplyEmailListItem
+import com.example.reply.ui.components.ReplyEmailThreadItem
+import com.example.reply.ui.components.ReplySearchBar
+import com.example.reply.ui.utils.ReplyContentType
+import com.example.reply.ui.utils.ReplyNavigationType
+
+@Composable
+fun ReplyInboxScreen(
+    contentType: ReplyContentType,
+    replyHomeUIState: ReplyHomeUIState,
+    navigationType: ReplyNavigationType,
+    closeDetailScreen: () -> Unit,
+    navigateToDetail: (Long, ReplyContentType) -> Unit
+) {
+    /**
+     * When moving from LIST page to LIST_AND_DETAIL page on screen expand, set first email selected as default
+     * When moving from LIST_AND_DETAIL page to LIST page clear the selection and user should see LIST screen.
+     */
+
+    LaunchedEffect(key1 = contentType) {
+        if (contentType == ReplyContentType.DUAL_PANE && replyHomeUIState.selectedEmail == null) {
+            replyHomeUIState.emails.firstOrNull()?.let { firstEmail ->
+                navigateToDetail.invoke(firstEmail.id, ReplyContentType.DUAL_PANE)
+            }
+        } else if (contentType == ReplyContentType.SINGLE_PANE && !replyHomeUIState.showDetailScreenOnly) {
+            closeDetailScreen.invoke()
+        }
```

Comment:
Nit: Looking at this some more, I think this is a composition loop that we can avoid with a bit more careful state handling.

The issue here is that `replyHomeUiState.selectedEmail` is doing two things: it is managing which email to show in the detail panel (if any) and it is also managing whether or not to show the detail.

I think if we split apart those issues and combine them back together, we can avoid the need for a `LaunchedEffect` entirely.

Let's say that we added a `isDetailOpen: Boolean` field to `replyHomeUiState`. This field would be set to `true` upon `navigateToDetail`, and set to `false` on `closeDetailScreen`.

We'd still have `selectedEmail`, but instead of starting at `null`, we could just set it as the first email in the list right out of the gate.

I think that would remove the need for `showDetailScreenOnly` as well, which feels a bit strange to have a `ViewModel` trying to know the state of the UI.

Reply:
Thanks for suggestion I removed the `selectedEmail` dependency and made first email as default. But we will still need LaunchEffect to close List/Detail combination  when moving from Single pane not selected to Double-pane and back.

https://user-images.githubusercontent.com/4903762/180475948-ea1f3a4b-ac97-4aae-932d-4f445b6d28c3.mov

Reply:
Got it, that usage of `LaunchedEffect` makes sense to me for that behavior.

Although taking a step back, is that desired? I feel like as a user I'd expect to stay in the same place upon rotating back if I don't interact with the app at all. Perhaps I rotated accidentally, or maybe I was closing my foldable (causing the app to appear on the front screen), and then upon reopening I want to return to where I was before.

Reply:
Tbh it's something I just followed Material large screen guidance, I can check with them what was the reason/research behind it. 
For me I can understand this flow seems correct -> I expand device I see detail without touching anything if fold back I should be same state as before (Original list).